File: src/main/org/cuair/ground/models/PlaneModel.java
package org.cuair.ground.models;

import java.sql.Timestamp;
import java.util.Objects;
import javax.validation.constraints.NotNull;
import javax.persistence.MappedSuperclass;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

/**
 * Base class for all settings that get sent to the plane. All State models should extend this
 * class. It consists of a timestamp indicating when it was created.
 */
@MappedSuperclass
public abstract class PlaneModel extends TimestampModel {

    /** A timestamp for the plane */
    @Temporal(TemporalType.TIMESTAMP)
    private Timestamp planeTimestamp;
    
    /**
     * Retrieves the plane timestamp
     *
     * @return Timestamp timestamp
     */
    public Timestamp getPlaneTimestamp() {
        return planeTimestamp;
    }
    
    /**
     * Sets the timestamp
     *
     * @param planeTimestamp new timestamp
     */
    public void setPlaneTimestamp(Timestamp planeTimestamp) {
        this.planeTimestamp = planeTimestamp;
    }
    
    @Override
    public boolean equals(@NotNull Object o) {
        if (o == null || !super.equals(o)) {
            return false;
        }
        PlaneModel other = (PlaneModel) o;
        
        return Objects.deepEquals(this.planeTimestamp, other.getPlaneTimestamp());
    }
}

File: src/main/org/cuair/ground/models/plane/PlaneModel.kt
package org.cuair.ground.models.plane

import java.sql.Timestamp
import java.util.Objects
import org.cuair.ground.models.TimestampModel
import javax.persistence.MappedSuperclass

/**
 * Base class for all settings that get sent to the plane. All State models should extend this
 * class. It consists of a timestamp indicating when it was created.
 */
@MappedSuperclass
abstract class PlaneModel : TimestampModel() {

    /** A timestamp for the plane */
    private var planeTimestamp: Timestamp? = null

    /**
     * Retrieves the plane timestamp
     *
     * @return Timestamp timestamp
     */
    fun getPlaneTimestamp(): Timestamp? {
        return planeTimestamp
    }

    /**
     * Sets the timestamp
     *
     * @param planeTimestamp new timestamp
     */
    fun setPlaneTimestamp(planeTimestamp: Timestamp?): Unit {
        this.planeTimestamp = planeTimestamp
    }

    override fun equals(o: Any?): Boolean {
        if (o == null || !super.equals(o)) {
            return false
        }
        val other: PlaneModel = o as PlaneModel

        return Objects.deepEquals(this.planeTimestamp, other.getPlaneTimestamp());
    }
}
